parsl/executors/workqueue/exec_parsl_function.py

```diff
@@ -0,0 +1,183 @@
+from parsl.app.errors import RemoteExceptionWrapper
+from parsl.data_provider.files import File
+from parsl.utils import get_std_fname_mode
+import traceback
+import sys
+import pickle
+
+# This scripts executes a parsl function which is pickled in a file:
+#
+# exec_parsl_function.py map_file function_file result_file
+#
+# map_file: Contains a pickled dictionary that indicates which local_paths the
+#           parsl Files should take.
+#
+# function_file: Contains a pickle parsl function.
+#
+# result_file: It will contain the result of the function, including any
+#              exception generated. Exceptions will be wrapped with RemoteExceptionWrapper.
+#
+# Exit codes:
+# 0: The function was evaluated to completion. The result or any exception
+#    wrapped with RemoteExceptionWrapper were written to result_file.
+# anything else: There was an error that prevented writing to the result file altogether.
+#                The exit code corresponds to whatever the python interpreter gives.
+#
+
+
+def load_pickled_file(filename):
+    with open(filename, "rb") as f_in:
+        return pickle.load(f_in)
+
+
+def dump_result_to_file(result_file, result_package):
+    with open(result_file, "wb") as f_out:
+        pickle.dump(result_package, f_out)
+
+
+def remap_location(mapping, parsl_file):
+    if not isinstance(parsl_file, File):
+        return
+    # Below we rewrite .local_path when scheme != file only when the local_name
+    # was given by the main parsl process.  This is the case when scheme !=
+    # 'file' but .local_path (via filepath) is in mapping.
+    if parsl_file.scheme == 'file' or parsl_file.local_path:
```

Comment:
I'm concerned (though I haven't tested this) that this will break with eg HTTPInTaskStaging, which sets local path ahead of time to mean "the file will appear here on the execute side, and a wrapper will later do that staging" rather than "the file is here on the submit side for workqueue to stage in". Fixing that should probably wait for another round of thinking about data staging architecture with the WQ staging model better incorporated.

Reply:
(turns out it doesn't...)

I'm not super happy about the parsl-level story for how this is working though but I'll leave that for future work.

Reply:
Got it. Work queue should be only remapping files that it staged. If it is not working, maybe it means that in some cases it is staging the wrong files. For example, if the .local_name set for execution also happens to exists at the main master process, but they refer to different contents. Is there enough information at the main parsl process to implement something like a `File.executor_should_stage()`?

Reply:
> Is there enough information at the main parsl process to implement something like a File.executor_should_stage()?

No. That's why I'm unhappy. it's not a workqueue problem.